drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,333 @@
+	static const uint8_t ws2812_uart_##idx##_color_mapping[] = DT_INST_PROP(idx, color_mapping)
+
+#define WS2812_NUM_COLORS(idx) (DT_INST_PROP_LEN(idx, color_mapping))
+
+/* Get the latch/reset delay from the "reset-delay" DT property. */
+#define WS2812_RESET_DELAY(idx) DT_INST_PROP_OR(idx, reset_delay, WS2812_DEFAULT_RESET_DELAY_MS)
+
+#define UART_NODE(idx) DT_PARENT(DT_INST(idx, worldsemi_ws2812_uart))
+
+#define WS2812_UART_DEVICE(idx)                                                                    \
+                                                                                                   \
+	BUILD_ASSERT(DT_PROP(UART_NODE(idx), tx_invert),                                           \
+		     "This driver depends on tx-invert UART property and "                         \
+		     "is not set. Please check your device-tree settings");                        \
+                                                                                                   \
+	BUILD_ASSERT(DT_PROP(UART_NODE(idx), data_bits), "data-bits property missing");            \
+                                                                                                   \
+	BUILD_ASSERT(DT_PROP(UART_NODE(idx), data_bits) <= 8, "data-bits > 8 is not supported");   \
+                                                                                                   \
+	BUILD_ASSERT((2 + DT_PROP(UART_NODE(idx), data_bits)) % DT_INST_PROP(idx, frame_len) == 0, \
+		     "data-bits+2 should be a multiple of frame-len");                             \
+                                                                                                   \
+	static uint8_t ws2812_uart_##idx##_px_buf[WS2812_UART_BUFSZ(                               \
+		idx, DT_INST_PROP(idx, rgb_frame_per_uart_frame))];                                \
+                                                                                                   \
+	WS2812_COLOR_MAPPING(idx);                                                                 \
+	static const uint8_t ws2812_uart_##idx##_lut[] = WS2812_LOOK_UP_TABLE_PREPARE(             \
+		DT_INST_PROP(idx, rgb_frame_per_uart_frame), DT_INST_PROP(idx, zero_frame),        \
+		DT_INST_PROP(idx, one_frame), DT_INST_PROP(idx, frame_len));                       \
+                                                                                                   \
+	static const struct ws2812_uart_cfg ws2812_uart_##idx##_cfg = {                            \
+		.uart_dev = DEVICE_DT_GET(DT_INST_BUS(idx)),                                       \
+		.px_buf = ws2812_uart_##idx##_px_buf,                                              \
+		.num_colors = WS2812_NUM_COLORS(idx),                                              \
+		.color_mapping = ws2812_uart_##idx##_color_mapping,                                \
+		.reset_delay = MAX(WS2812_RESET_DELAY(idx), 8),                                    \
```

Comment:
and this doesn't even make sense, the default above is 50, but here it is being capped at 8? How does that work?